### PR TITLE
Add Deployments tab, do not in-place replace runtime console at Applications

### PR DESCRIPTION
**Pull request by andrewazores**

### CODE REVIEW

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Comment by joshuawilson** [comment]
[test]

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-6 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Comment by andrewazores** [comment]
@jiekang mind taking a look?

File: src/app/layout/header/menus.service.ts [hunk 1]
```diff
@@ -111,4 +86,37 @@ export class MenusService {
...
+    if (displayDeployments) {
+      menus.push({
+        name: 'Deployments',
+        path: 'deployments'
+      });
+    }
+    menus.push({
+      name: 'Environments',
+      path: 'environments'
+    });
```
Comment (jiekang) [inline src/app/layout/header/menus.service.ts]: Trivial nit:

Deployments will probably eventually replace both Applications and Environments and I feel like the order doesn't matter. Putting Deployments last will drop the need to have an extra push after the if statement.

File: src/app/space/create/create-routing.module.ts [hunk 2]
```diff
@@ -39,6 +39,17 @@ const routes: Routes = [
           featureName: 'Applications'
         }
       },
+      {
+        path: 'deployments',
```
Comment (jiekang) [inline src/app/space/create/create-routing.module.ts]: By having the menus display deployments conditionally and the routing always have deployments available, the page is can be visited in production by manually accessing the url [...]/create/deployments

Is there no way to drop the route conditionally as well?

Comment (andrewazores) [reply src/app/space/create/create-routing.module.ts]: It's possible, but means there will either need to be some utility extracted out somewhere up the file hierarchy, or some duplicated logic to check for development/production. I asked @joshuawilson if it was sufficient to just visually hide it and leave the route in place and IIRC he was okay with that.

File: src/app/layout/header/menus.service.ts (hunk 1 above)
Comment (andrewazores) [reply src/app/layout/header/menus.service.ts]: Yea, true. I'll touch that up.

File: src/app/space/create/create-routing.module.ts (hunk 2 above)
Comment (jiekang) [reply src/app/space/create/create-routing.module.ts]: Okay. If Joshua is fine then I have no objections either.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-8 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

File: src/app/space/create/create-routing.module.ts (hunk 2 above)
Comment (joshuawilson) [reply src/app/space/create/create-routing.module.ts]: I'm fine with having the routing in place and just hiding the menu option. The main thing is to not break production. A hidden route while under development sounds like a reasonable option.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-9 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-10 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2252-13 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2252-fabric8-ui.badger.fabric8.io

**Pull request merged**